### ISSUE TITLE
[Feature] Create installer for windows

### ISSUE THREAD

**Issue by zodman** [mattermost/desktop#103]
Thanks!

**Comment by jeremycook** [comment]
I would love a windowed installer, but if it would be awhile before that could be finished then I would not be opposed to batch file installation on Windows. Two batch files could be created:

* SingleUserSetup.bat - Installs to current user's AppData folder and adds shortcuts to user's start menu and taskbar.
* MultipleUserSetup.bat - Installs to Program Files... and adds shortcuts to every user's start menu and taskbar.

If there is interest, I have a batch file that already does something similar to SingleUserSetup.bat and could provide that as a starting point, or do it via pull request. Just let me know.

**Comment by bakoonin** [comment]
Used the unicode version of [Inno Setup 5](http://www.jrsoftware.org/isdl.php) (free as in no money) to create an installation package for Mattermost Desktop 1.1.1.

I extracted the Mattermost Desktop files to the `FILES` directory, and put my optional config files (`bounds-info.json` and `config.json`) in the `CONFIG` folder.

In the Inno Setup script below I've commented out the config files. I only include them in the install package meant for home users, to make sure Things Just Work. In an enterprise environment you'll probably use GPO/GPP and security groups to roll out configs.

I also commented out the registry key to autostart Mattermost Desktop on login, which is also meant for home users. At work we have multiple users on each workstation, so global autostart would be an annoyance. For enterprise, use autostart per user (HKU regkeys + security groups) through GPO/GPP.

MattermostDesktop.iss:

```
[Setup]
AppName=Mattermost Desktop
AppVerName=Mattermost Desktop
AppVersion=1.1.1
DefaultDirName={pf}\MattermostDesktop
DefaultGroupName=Mattermost Desktop
UninstallDisplayIcon={app}\Mattermost.exe
Compression=lzma2/fast
SolidCompression=no
OutputBaseFilename=MattermostSetup_1.1.1
ArchitecturesAllowed=x64
ArchitecturesInstallIn64BitMode=x64

[Files]
Source: "FILES\*"; DestDir: "{app}"; Flags: recursesubdirs
;Source: "CONFIG\*.json"; DestDir: "{userappdata}\Mattermost"; Flags: onlyifdoesntexist uninsneveruninstall

[Icons]
Name: "{group}\Mattermost Desktop"; Filename: "{app}\Mattermost.exe"
Name: "{commondesktop}\Mattermost Desktop"; Filename: "{app}\Mattermost.exe"

;[Registry]
;Root: HKLM; Subkey: "Software\Microsoft\Windows\CurrentVersion\Run"; ValueType: string; ValueName: "MattermostDesktop"; ValueData: "{app}\Mattermost.exe"; 
```
Tested on the 64-bit versions of Windows 7 Enterprise and Windows 8.1 Pro. If you want to support x86 you'll need to change things.

**Comment by yuya-oc** [comment]
Sorry for late. I will consider using `electron-builder` advised at #105 . Certainly InnoSetup also meets this, but it's Windows application. So now I think it might be a little hard to introduce into our CI.

**Comment by kwill** [comment]
When packaging for Windows, please consider Chocolatey as a Windows distribution channel. (Separate ticket?)

Chocolatey is a distribution tool for Windows applications, including many free software applications (for example, https://chocolatey.org/packages/pidgin). It allows an end-user to get application updates somewhat like using apt or yum on GNU/Linux.

https://github.com/chocolatey/choco/wiki/ChocolateyFAQs
https://github.com/chocolatey/choco/wiki/Moderation

**Comment by yuya-oc** [comment]
@kwill Yeah, please separate the issue.

**Comment by Linkletter** [comment]
This is a big one. We are piloting Mattermost at my university and installation of desktop apps is slowing down adoption. As-is I think most people will run the app from their Downloads folder, and very few will set it to start at login. There is a lot of potential to enhance learning, particularly in online or blended courses, but it's hard to assess this without higher adoption. I think an installer would help a great deal.

**Comment by Razzeee** [comment]
Fixed in master since 68b54ee4aa122ff4d01999abbaf03b8bf55c9ff6
